### PR TITLE
Feature/proxy authentication for newman CLI 

### PR DESCRIPTION
**Pull request by someshkoli**
Fixes Issue / Feature request : #1599

As the issue stated there was no other way to use newman  CLI in a proxy environment rather than setting proxy authentication credentials in the environment. This is not so secure in some cases. To avoid this one way is to supply proxy credentials via CLI as an option to the newman. 

This PR provides this solution by supplying proxy authentication credentials via CLI as show below
`newman run sample-collection.json --proxy username:password`

Running collection without proxy set:

withouproxy

Running collection with proxy set:
with proxy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -494,7 +494,7 @@ Several members of the Postman community have created custom reporters offering
 Once the custom reporter NPM package has been installed either globally or locally, this can be then used with Newman in the following ways:
 
 ```console
-$ newman run /path/to/collection.json -r htmlextra,csv
+$ newman run /path/to/collection.json -r htmlextra.csv
```
Comment (shamasis) [inline README.md]: This isn’t an error. -r htmlextra,csv implies using two reporters.

**Comment by shamasis** [comment]
@codenirvana @codeitva - can you point to the right runtime proxy input code / doc?

File: README.md (hunk 1 above)
Comment (someshkoli) [reply README.md]: Oh damn. I totally miss  read that. My bad

**Comment by someshkoli** [comment]
Okay ill check for the runtime options and push the proxy to them directly. Also provide option to specity proxy authentication method

**Comment by someshkoli** [comment]
@shamasis  What all proxy authentication methods should be added?  Also in the sdk(postman-collection)  there is no way to specify the authentication method. so the default which postman is using right now is basic.

**Comment by shamasis** [comment]
These are options passed to runtime when a new run instance is created.

**Comment by someshkoli** [comment]
> These are options passed to runtime when a new run instance is created.

Nah i was talking about how the user will supply these values to newman

**Comment by shamasis** [comment]
That’s the whole reason we didn’t reinvent the wheel by adding in CLI and instead left it as all other CLI tools do ... using environment vars!

I wouldn’t over complicate this feature. I would either put clean CLI options or not build any CLI options. 😛 JSON source will be an overkill.

**Comment by someshkoli** [comment]
xD okay then CLI options it is. Also could you tag someone who could help with specifying auth. I'm having trouble in pin pointing where to add the auth options.Thanks

**Comment by someshkoli** [comment]
@shamasis can you please review this ?

**Review by codenirvana — COMMENTED** [review]
@someshkoli Great work but we are not planning to support this option at the moment.

Currently, one can configure proxy using `HTTP_PROXY`/`HTTPS_PROXY` environment variables and we haven't seen a lot of users requesting this so closing this for now.

**Comment by swatig90** [comment]
Hi @someshkoli , I am trying to execute some requests which are running behind proxy and facing exactly same issue which is described here and also issue 1599.
Could you please let me know whether sending proxy along with newman run command is possible now, or if there is any other alternative.

I already tried --env-var to set HTTP_PROXY/ HTTPS_PROXY, but it dint help.

Error message I receive is: tunneling socket could not be established, statusCode=403

Many thanks in advance!

**Comment by someshkoli** [comment]
> Hi @someshkoli , I am trying to execute some requests which are running behind proxy and facing exactly same issue which is described here and also issue 1599.
> Could you please let me know whether sending proxy along with newman run command is possible now, or if there is any other alternative.
> 
> I already tried --env-var to set HTTP_PROXY/ HTTPS_PROXY, but it dint help.
> 
> Error message I receive is: tunneling socket could not be established, statusCode=403
> 
> Many thanks in advance!

Yeah you can do that, but you need to pass that var to your shell's env not as a flag
You can easily do this by temporarily passing them using `export` on linux shell or permanently storing them in your shell's rc file and equv in windows too

**Comment by vbisbest** [comment]
I really wish this would be accepted.  Using environment variables for this can have consequences.  What about after the run and removing the variables?  Chrome uses the same environment variables and you may not want chrome to use a proxy.   CLI option is the cleanest way to do this especially when automating newman.

**Comment by swatig90** [comment]
Suggested solution by Someshkoli had worked well for me. This is how I implemented. I setup HTTP_PROXY and HTTPS_PROXY environment variable.
Another alternative is setup proxy variables at npm level.

Chrome will not be impacted via any of these (it dint at least in my case), however access to other APIs which do not require proxy, might be impacted.

**Pull request closed without merging**